common/lib/capa/capa/responsetypes.py

```diff
@@ -481,6 +481,16 @@ def get_hints(self, student_answers, new_cmap, old_cmap):
                 'old_cmap_dict': old_cmap.get_dict(),
             }
 
+            # We allow context-specific overrides for limits in the safe executation environment.
+            # If available, we use this response's stringified course key to identify the set
+            # of overrides we want to use, allowing us to tweak limits for particular course
+            # runs via Django settings.
+            # If the course key is not available, fall back to None (no overrides).
+            try:
+                limit_overrides_context = str(self.capa_module.scope_ids.usage_id.course_key)
+            except AttributeError:
+                limit_overrides_context = None
+
```

Comment:
Yes, this is line-for-line identical with what's in `capa_problem.py`. I considered extracting this out, but I couldn't think of a useful abstraction. I considered doing something like this in safe_exec.py, but wasn't sure it was worth it:

```python
def get_limit_overrides_for_block(block):
   """
   ...
   """
   try:
       return str(self.scope_ids.usage_id.course_key)
   except AttributeError:
       return None
```
Either way, it looks like I'm introducing edx-platform concepts in CAPA code. I'm not sure how much we care about that boundary breakage. Open to any input here.

Reply:
We're leaking opaque-keys into capa, which I don't think is quite as bad. I'm fine with both the leakage and the minor duplication. One word of warning that this _might_ raise a `TypeError` in Old Mongo courses. That style of UsageKey doesn't have course run information, so the CourseKey portion of it will be incomplete. Any attempt to serialize the CourseKey would throw that error:

```
>>> from opaque_keys.edx.keys import CourseKey, UsageKey
>>> usage_id = "i4x://edX/DemoX.1/problem/466f474fa4d045a8b7bde1b911e095ca"
>>> usage_key = UsageKey.from_string(usage_id)
>>> usage_key.course_key
CourseLocator('edX', 'DemoX.1', None, None, None)
>>> str(usage_key.course_key)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/opaque_keys/__init__.py", line 174, in __str__
    return self._to_deprecated_string()
  File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/opaque_keys/edx/locator.py", line 368, in _to_deprecated_string
    return u'/'.join([self.org, self.course, self.run])
TypeError: sequence item 2: expected str instance, NoneType found
```

The reason I say might is because we generally try to use `map_into_course` to force run information into the keys when we're doing courseware rendering. In any case, also catching a `TypeError` probably wouldn't hurt.